.github/workflows/vitest.yml

```diff
@@ -0,0 +1,52 @@
+name: Vitest
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '24 21 * * 4'
+
+jobs:
+    vitest_run:
+        name: run vitest
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            security-events: write
+            actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+        steps:
+          - name: Checkout code
+            uses: actions/checkout@v3
+
+          - name: Install Vitest
+            working-directory: ./client
+            run: |
+              npm install
+
+          - name: Run Vitest
+            working-directory: ./client
+            run: npm test
```

Comment:
Couldn't you do `npm run coverage` as the next step and avoid the following job altogether?